lib/src/front_end/piece_factory.dart

```diff
@@ -696,11 +832,23 @@ mixin PieceFactory {
   /// If [splitBeforeOperator] is `true`, then puts [operator] at the beginning
   /// of the next line when it splits. Otherwise, puts the operator at the end
   /// of the preceding line.
+  ///
+  /// If [allowInnerSplit] is `true`, then a newline inside the target or
+  /// right-hand side doesn't force splitting at the operator itself.
   Piece createAssignment(
       AstNode target, Token operator, Expression rightHandSide,
       {bool splitBeforeOperator = false,
       bool includeComma = false,
-      bool spaceBeforeOperator = true}) {
+      bool spaceBeforeOperator = true,
+      bool allowInnerSplit = false}) {
+    // If the right-hand side can have block formatting, then a newline in
+    // it doesn't force the operator to split, as in:
+    //
+    //    var list = [
+    //      element,
+    //    ];
+    allowInnerSplit |= rightHandSide.canBlockSplit;
```

Comment:
I don't have a problem with this but I don't recall how common it is - do we consider it idiomatic to use bitwise-or-assignment for booleans?

Reply:
It's not as common as I expected, but it happens. Using `||=` would be the idiomatic self-assignment operator for this, but Dart doesn't have `||=` and `|=` does the right thing since `bool` implements `|`.